system/lib/pthread/library_pthread_stub.c

```diff
@@ -285,11 +285,11 @@ int pthread_attr_destroy(pthread_attr_t *attr) {
   return 0;
 }
 
-int pthread_setcancelstate() {
```

Comment:
Why did the compiler allow this in the first place?  Are we missing a header in include here?

Reply:
In C prior to C23, this is K&R-style function definition since it is missing `void` in the parameter location. Which I guess means it matches with any declaration? It's kind of weird because I would have expected the non-K&R declaration to force the definition to have the proper type, but idk exactly how this works.